app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt

```diff
@@ -321,7 +321,12 @@ class ParkRepositoryFirestore(private val db: FirebaseFirestore, testing: Boolea
     return try {
       val pid = document.id
       val name = document["name"] as? String ?: ""
-      val location = document["location"] as? ParkLocation ?: ParkLocation(0.0, 0.0, "0")
+      val locationMap = document["location"] as? Map<*, *> ?: emptyMap<String, Any>()
```

Comment:
Question: I think I am missing something: what is the difference between this version of the code and the last one?

Reply:
Essentially, the previous version is just incorrect: parklocations are stored as `Map` values in the firestore database so trying to naively cast it to ParkLocation when fetching from the database results in a null value (since what's fetched is not a `Parklocation`) so i end up with parks that just have the default `ParkLocation(0.0, 0.0, "0")`

This is the reason the markers were not properly displayed in the event overview map to begin with
Thank you for your inquiry and let me know if this is not clear enough!